client/state/current-user/selectors.js

```diff
@@ -44,6 +44,21 @@ export function getCurrentUserLocale( state ) {
 }
 
 /**
+ * Returns the number of sites for the current user.
+ *
+ * @param  {Object}  state  Global state tree
+ * @return {?Number}        Current user site count
+ */
+export function getCurrentUserSiteCount( state ) {
+	const user = getCurrentUser( state );
+	if ( ! user ) {
+		return null;
+	}
+
+	return user.site_count || null;
```

Comment:
Agreed! Fixed in https://github.com/Automattic/wp-calypso/pull/14022/commits/ef650dfd7aae2ba7ec9df10dc3bcc3f09e067143